dbfit-java/core/src/main/java/dbfit/util/DbParameterAccessor.java

```diff
@@ -74,20 +74,27 @@ public void bindTo(StatementExecution cs, int ind) throws SQLException{
         this.cs=cs;
         this.index=ind;    
         if (direction != INPUT){
-            cs.registerOutParameter(ind, getSqlType());
+            cs.registerOutParameter(ind, getSqlType(), direction == RETURN_VALUE);
         }
     }
 
     public void set(Object value) throws Exception {
         if (direction == OUTPUT|| direction == RETURN_VALUE)
             throw new UnsupportedOperationException("Trying to set value of output parameter "+name);
+
+        if (value != null) {
+            TypeNormaliser tn = TypeNormaliserFactory.getNormaliser(value.getClass());
+            if (tn != null) {
+                value = tn.normalise(value);
+            }
+        }
         cs.setObject(index, value, sqlType, userDefinedTypeName);
     }    
 
     public Object get() throws IllegalAccessException, InvocationTargetException {
         try{
             if (direction.equals(INPUT))
-                throw new UnsupportedOperationException("Trying to get value of input parameter "+name);            
+                throw new UnsupportedOperationException("Trying to get value of input parameter "+name);
```

Comment:
Maybe would be cleaner to post these formatting/whitespace changes as separate PR? (There are other pieces in this class to tidy up, we can revise the whole file).